src/librustc/ty/layout.rs

```diff
@@ -523,8 +523,12 @@ pub type FieldPath = Vec<u32>;
 /// A structure, a product type in ADT terms.
 #[derive(PartialEq, Eq, Hash, Debug)]
 pub struct Struct {
+    /// Maximum alignment of fields and repr alignment.
     pub align: Align,
 
+    /// Alignment of struct fields, for comparison with LLVM type.
+    pub abi_align: Align,
```

Comment:
We shouldn't be talking about LLVM in `librustc`, ideally. How about `primitive_align`? That is, the alignment obtained from primitives' alignment, without any user annotations.